React/Views/ScrollView/RCTScrollView.m

```diff
@@ -324,11 +325,28 @@ - (void)_keyboardWillChangeFrame:(NSNotification *)notification
   }
 
   CGPoint newContentOffset = _scrollView.contentOffset;
-  CGFloat contentDiff = endFrame.origin.y - beginFrame.origin.y;
-  if (self.inverted) {
-    newContentOffset.y += contentDiff;
+  UIResponder *firstResponder = [UIResponder currentFirstResponder];
+  if ([firstResponder isKindOfClass: [UITextField class]] && [(UITextField *) firstResponder isDescendantOfView:_scrollView]) {
+    UITextField *textField = [UIResponder currentFirstResponder];
+    CGRect textFieldFrame = [textField.superview convertRect:textField.frame toView:nil];
+    // [32] is a placeholder. It's the distance between the bottom of the textfield and the top of the keyboard
+    // What value should be used? Should it be customizable with a prop?
+    CGFloat textFieldBottom = textFieldFrame.origin.y + textFieldFrame.size.height + 32;
```

Comment:
> For android (Java/Kotlin) we can use; [`WindowInsets`](https://developer.android.com/develop/ui/views/layout/sw-keyboard#java) api to get the height of the soft. keyboard

UIKit seems to have api's like [view.keyboardLayoutGuide](https://developer.apple.com/documentation/uikit/keyboards_and_input/adjusting_your_layout_with_keyboard_layout_guide)
and,
[UIKeyboardLayoutGuide](https://developer.apple.com/documentation/uikit/uikeyboardlayoutguide)

PS: Providing prop to adjust it manually seems nice customization but, using native api's to calculate the height would be more appropriate in this case.